core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -1374,35 +1374,44 @@ class KafkaApis(val requestChannel: RequestChannel,
     val initPidRequest = request.body[InitPidRequest]
     val transactionalId = initPidRequest.transactionalId
 
-    // Send response callback
-    def sendResponseCallback(result: InitPidResult): Unit = {
-      def createResponse(throttleTimeMs: Int): AbstractResponse = {
-        val responseBody: InitPidResponse = new InitPidResponse(throttleTimeMs, result.error, result.pid, result.epoch)
-        trace(s"InitPidRequest: Completed $transactionalId's InitPidRequest with result $result from client ${request.header.clientId}.")
-        responseBody
+    if(authorize(request.session, Write, new Resource(ProducerTransactionalId, transactionalId))) {
```

Comment:
In fact, we also need to authorize the ProducerIdResource in the ProduceRequest. We can't actually authorize the PID used, but at least this lets users forbid some producers from using _any_ PID.